src/frequenz/client/reporting/__main__.py

```diff
@@ -93,7 +98,7 @@ async def run(
     metric_names: list[str],
     start_dt: datetime,
     end_dt: datetime,
-    resolution: int,
+    resampling_period: timedelta,
```

Comment:
You are passing an int (and in fact also `None` is possible, which is missing here). I would leave this and convert to timedelta (if set) below.

Reply:
Sorry, I am not entirely sure what you mean with this comment. I tried the following change:

Reply:
Here in the `run` method, the type should be `int | None`, not in the `ReportingApiClient` Constructor if I read it correctly.
You then only need to convert it to timedeltta in line 144. 

I would also recommend renaming the parameter to `resampling_period_s` to indicate that it is seconds in the name right away. Once it is a timedelta that extra `_s` won't be needed anymore

Reply:
>Here in the run method, the type should be int | None

Yes, and below:

```
resampling_period = timedelta(seconds=resampling_period_s) if resampling_period_s is not None or None
return client.list_single_component_data(
            microgrid_id=microgrid_id,
            component_id=component_id,
            metrics=metrics,
            start_dt=start_dt,
            end_dt=end_dt,
            resampling_period=resampling_period,
            include_states=states,
            include_bounds=bounds,
        )
```